docs/proposals/admission_control_changes.md

```diff
@@ -0,0 +1,279 @@
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Admission control changes
+
+We are proposing a series of changes to admission control in an attempt to make
+it less error-prone, easier to configure, and more flexible for admission plugin
+developers.
+
+
+## Hard-coded plugin execution order
+
+It's unlikely that admins need to change the order in which admission plugins
+are executed.  Furthermore, running certain plugins "out of order" could result
+in undesirable or incorrect behavior. For example, if you run `ResourceQuota`
+before `LimitRanger`, you could end up incorrectly calculating quota because
+it's possible for `LimitRanger` to modify pod values after `ResourceQuota`
+performed its quota checks.
+
+Because of this, we want to move to a model where admission plugin execution
+order is hard-coded. We would therefore deprecate the `--admission-control`
+setting from `pkg/genericapiserver/options.ServerRunOptions`.
+
+
+## One config file per plugin
+
+All admission plugins currently share a single config file, specified via
+`--admission-control-config-file`. This is problematic if multiple plugins
+decide they want to use the same names for their configuration sections. For
+example, if plugin `Foo` wants
+
+```yaml
+myConfig:
+  someField: red
+```
+
+and plugin `Bar` wants
+
+```yaml
+myConfig:
+  someField:
+  - one
+  - two
+  - three
+```
+
+this won't work because of the naming collision and the type of `someField` is
+different for the two plugins. What we'd like to see instead is:
+
+1. deprecate `--admission-control-config-file`
+1. replace it with `--admission-plugin PLUGIN=CONFIG_FILE` (or an equivalent
+   setup in a structured configuration, once we've fully moved over)
+
+You would specify `--admission-plugin` once per plugin that you want to
+configure.
+
+
+## Ability to enable/disable individual plugins
+
+We want to hard-code a set of plugins that are enabled by default.
+Administrators need to be able to override the defaults, of course. To simply
+enable or disable a specific plugin, you would create a config file that looks
+like this:
+
+```yaml
+kind: DefaultAdmissionConfig
+version: v1
+enabled: false
+```
+
+The flow to determine if a plugin is enabled would look like this:
+
+1. If the plugin is on by default
+  1. If there is no config file for it, it remains enabled
+  1. Otherwise (have config file)
+    1. If it deserializes to `DefaultAdmissionConfig`, it uses the value of
```

Comment:
For consistency it would be nice if it was always yaml/json, but I don't know that we need to require it.